examples/webgl_per_vertex_ao.html

```diff
@@ -0,0 +1,450 @@
+
+						} else { // j is last index in face
+
+							edge1Index = indices[ j - 1 ];
+							edge2Index = indices[ j - 2 ];
+
+						}
+
+						edge1.set(
+							positions[ pIndex * 3 ] - positions[ edge1Index * 3 ],
+							positions[ pIndex * 3 + 1 ] - positions[ edge1Index * 3 + 1 ],
+							positions[ pIndex * 3 + 2 ] - positions[ edge1Index * 3 + 2 ]
+						);
+
+						edge2.set(
+							positions[ pIndex * 3 ] - positions[ edge2Index * 3 ],
+							positions[ pIndex * 3 + 1 ] - positions[ edge2Index * 3 + 1 ],
+							positions[ pIndex * 3 + 2 ] - positions[ edge2Index * 3 + 2 ]
+						);
+
+						areas[ pIndex ] += 0.5 * edge1.cross( edge2 ).length();
+
+					} )
+
+					areas[ pIndex ] /= 3;
+
+				} );
+
+				geometry.addAttribute( 'area', new THREE.Float32BufferAttribute( areas, 1 ) );
+
+			}
+
+			function ambientOcclusion( geometry ) {
+
+				var colors = [];
+
+				var positions = geometry.attributes.position.array;
+				var normals = geometry.attributes.normal.array;
+				var bentNormals = normals.slice();
+				var areas = geometry.attributes.area.array;
+
+				var receiver = new THREE.Vector3();
+				var receiverNormal = new THREE.Vector3();
+
+				var emitter = new THREE.Vector3();
+				var emitterNormal = new THREE.Vector3();
+
+				var bentNormal = new THREE.Vector3();
+
+				var v = new THREE.Vector3();
+
+				function computeVertexOcclusion( i ) {
+
+					var value = 0;
+
+					for ( var j = 0; j < positions.length; j += 3 ) {
+
+						var emitterArea = areas[ j / 3 ];
+
+						emitter.fromArray( positions, j );
+						emitterNormal.fromArray( normals, j );
+
+						v.subVectors( emitter, receiver )
+
+						var d2 = v.dot( v ) + 1e-16;
+
+						if( d2 < 4 * emitterArea) {
+
+							v.multiplyScalar( 1 / Math.sqrt( d2 ) ); // normalise
+
+							value += solidAngle( v, d2, receiverNormal, emitterNormal, emitterArea );
+
+							bentNormal.sub( v.multiplyScalar( value ) );
+
+						}
+
+					}
+
+					bentNormal.normalize();
+					return saturate( 1 - value );
+
+				}
+
+				// loop over each vertex
+				for ( var i = 0; i < positions.length; i += 3 ) {
+
+					receiver.fromArray( positions, i );
+					receiverNormal.fromArray( normals, i );
+
+					if( bentNormals[ i ] ) receiverNormal.fromArray( bentNormals, i );
+					else bentNormal.copy( receiverNormal ); // initialise
+
+					// initialise to 1
+					if( !colors[ i / 3 ] ) colors[ i / 3 ] = 1;
+
+					var value = computeVertexOcclusion( i );
+
+					bentNormal.toArray( bentNormals, i );
+
+					colors[ i / 3 ] *= value;
+
+				}
+
+				// ignoring bent normals for now since with a single pass they don't improve appearance of the current model
+				// geometry.attributes.normal.set( bentNormals );
+
+				var vertexColors = [];
+				for( var k = 0; k < colors.length; k ++ ) vertexColors.push( colors[ k ], colors[ k ], colors[ k ] );
+
+				geometry.addAttribute( 'color', new THREE.BufferAttribute( new Float32Array( vertexColors ), 3 ) );
+
+				}
```

Comment:
Minor formatting issue. One tab too much.

Reply:
Linter errors should be fixed now.